lib/improver/ensemble_copula_coupling/ensemble_copula_coupling.py

```diff
@@ -622,22 +622,99 @@ def process(self, forecast_probabilities, no_of_percentiles=None,
+                define the shape of the distribution. For the truncated normal
+                distribution, the shape parameters should be appropriate for
+                the mean and standard deviation used to describe the
+                distribution.
+                Please note that for use with
+                :meth:`~improver.ensemble_calibration.ensemble_calibration.\
+ContinuousRankedProbabilityScoreMinimisers.calculate_truncated_normal_crps`,
+                the shape parameters for a truncated normal distribution with
+                a lower bound of zero should be [0, np.inf].
+
         """
-        pass
+        try:
+            self.distribution = getattr(stats, distribution)
+        except AttributeError as err:
+            msg = ("The distribution requested {} is not a valid distribution "
+                   "in scipy.stats. {}".format(distribution, err))
+            raise AttributeError(msg)
+
+        if shape_parameters is None:
+            shape_parameters = []
+        self.shape_parameters = shape_parameters
+
+    def __repr__(self):
+        """Represent the configured plugin instance as a string."""
+        result = ('<FromMeanAndVariance: distribution: {}; '
+                  'shape_parameters: {}>')
+        return result.format(
+            self.distribution.name, self.shape_parameters)
+
+    def _rescale_shape_parameters(self, mean, std):
+        """
+        Rescale the shape parameters for the desired mean and standard
+        deviation for the truncated normal distribution. The shape parameters
+        for any other distribution will remain unchanged.
+
+        For the truncated normal distribution, if the shape parameters are not
+        rescaled, then :data:`scipy.stats.truncnorm` will assume that the shape
+        parameters are appropriate for a standard normal distribution. As the
+        aim is to construct a distribution using specific values for the mean
+        and standard deviation, the assumption of a standard normal
+        distribution is not appropriate. Please see
+        :data:`scipy.stats.truncnorm` for some further information.
+
+        Args:
+            mean (numpy.ndarray):
+                Mean to be used to scale the shape parameters.
+            std (numpy.ndarray):
+                Standard deviation to be used to scale the shape parameters.
+
+        """
+        if self.distribution.name == "truncnorm":
+            if self.shape_parameters:
+                rescaled_values = []
+                for value in self.shape_parameters:
+                    rescaled_values.append((value - mean) / std)
+                self.shape_parameters = rescaled_values
+            else:
+                msg = ("For the truncated normal distribution, "
+                       "shape parameters must be specified.")
+                raise ValueError(msg)
+
+
+class GeneratePercentilesFromMeanAndVariance(BasePlugin, FromMeanAndVariance):
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Class name "GeneratePercentilesFromMeanAndVariance" doesn't conform to '((Test_[a-zA-Z0-9_]+)|([A-Z_][a-zA-Z0-9_]{2,30}))$' pattern](https://app.codacy.com/app/metoppv_tech/improver/pullRequest?prid=4540588)